nix/overlay.nix

```diff
@@ -1,8 +1,20 @@
 self: pkgs:
-{
+
+let
+  sources = import ./sources.nix;
+in {
   # packages defined in this repo
   hercules-ci-agent-packages = pkgs.callPackages ./packages.nix {};
 
   # overrides
   hercules-ci-agent = pkgs.haskell.lib.justStaticExecutables self.hercules-ci-agent-packages.hercules-ci-agent;
+
+  devTools = {
```

Comment:
Sorry, we can't add `devTools` here because this overlay is part of the public interface.

Reply:
That has always been a controversial topic :) 

Adding another overlay is also public interface that someone might use, but I think stating what is or is not the public interface will set the expectations about backwards compatibility.

I can add another overlay if you want, but I really think it's not worth the abstraction cost since noone will ever care.

Reply:
I don't think `module.nix` and friends should add `devTools` to `pkgs`. That just doesn't make sense. Can we reduce the scope, just restore it and fix it properly later? There's also more important stuff like the the NixOS `nix.package` and our pinned `nix` that need to be equivalent/compatible. We'll probably want have to use `overlay.nix` for that as well.
Issue: https://github.com/hercules-ci/hercules-ci-agent/issues/67

Reply:
If we revert, it will break Darwin support, that's also a public api in nix-darwin.

Reply:
So in that case it's better to leave `devTools` factoring for later if you want to move this forward.

Reply:
We only need to revert the part that merges devTools into the overlay iirc. I don't think darwin depends on devTools.

Reply:
Alright, will push this shortly.